hera_qm/utils.py

```diff
@@ -722,6 +726,8 @@ def apply_yaml_flags(uv, a_priori_flag_yaml, lat_lon_alt_degrees=None, telescope
         raise NotImplementedError("apply_yaml_flags does not support multiple spws at this time.")
     # if UVCal provided and lst_array is None, get lst_array from times.
     # If lat_lon_alt is not specified, try to infer it from the telescope name, which calfits files generally carry around
+    if unflag_first:
+        uv.flag_array[:] = False
```

Comment:
I think this is fine for any dimensions, but I'd recommend `uv.flag_array = np.zeros_like(uv.flag_array, dtype=bool)` to be more explicit.

Reply:
Done.